src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSpark.java

```diff
@@ -0,0 +1,440 @@
+
+            final byte[] insertedSequenceBytes = Arrays.copyOfRange(sequenceCopy, insertionStart - 1, insertionEnd);
+            if (previous.referenceInterval.getStart() > current.referenceInterval.getStart()) {
+                SequenceUtil.reverseComplement(insertedSequenceBytes, 0, insertedSequenceBytes.length);
+            }
+            insertedSequence = new String(insertedSequenceBytes);
+        }
+        return insertedSequence;
+    }
+
+    private static String getHomology(final AlignmentRegion current, final AlignmentRegion previous, final byte[] sequenceCopy) {
+        String homology = "";
+        if (previous.endInAssembledContig >= current.startInAssembledContig) {
+            final byte[] homologyBytes = Arrays.copyOfRange(sequenceCopy, current.startInAssembledContig - 1, previous.endInAssembledContig);
+            if (previous.referenceInterval.getStart() > current.referenceInterval.getStart()) {
+                SequenceUtil.reverseComplement(homologyBytes, 0, homologyBytes.length);
+            }
+            homology = new String(homologyBytes);
+        }
+        return homology;
+    }
+
+    private static boolean currentAlignmentRegionIsTooSmall(final AlignmentRegion current, final AlignmentRegion next, final Integer minAlignLength) {
+        return current.referenceInterval.size() - current.overlapOnContig(next) < minAlignLength;
+    }
+
+    @VisibleForTesting
+    static boolean treatNextAlignmentRegionInPairAsInsertion(AlignmentRegion current, AlignmentRegion next, final Integer minAlignLength) {
+        return treatAlignmentRegionAsInsertion(next) ||
+                (next.referenceInterval.size() - current.overlapOnContig(next) < minAlignLength) ||
+                current.referenceInterval.contains(next.referenceInterval) ||
+                next.referenceInterval.contains(current.referenceInterval);
+    }
+
+    private static boolean treatAlignmentRegionAsInsertion(final AlignmentRegion next) {
+        return next.mapqual < 60;
+    }
+
+    private static void writeVariants(final String fastaReference, final JavaRDD<VariantContext> variantContexts, final PipelineOptions pipelineOptions, final String outputPath) {
+
+        final List<VariantContext> variants = variantContexts.collect();
+        final List<VariantContext> sortedVariantsList = new ArrayList<>(variants);
+
+        final ReferenceMultiSource referenceMultiSource = new ReferenceMultiSource(pipelineOptions, fastaReference, ReferenceWindowFunctions.IDENTITY_FUNCTION);
+        final SAMSequenceDictionary referenceSequenceDictionary = referenceMultiSource.getReferenceSequenceDictionary(null);
+
+        sortedVariantsList.sort((VariantContext v1, VariantContext v2) -> IntervalUtils.compareLocatables(v1, v2, referenceSequenceDictionary));
+
+        log.info("Called " + variants.size() + " inversions");
+        final VCFHeader header = getVcfHeader(referenceSequenceDictionary);
```

Comment:
I don't really see how adding a stream and collect call makes things any clearer so I'm going to leave this as is. Plus I prefer to leave calls to `collect` on an RDD on their own line so that you can explicitly see the transition from RDD-land to Java-collection-on-the-driver land.